src/Gremlin.Net.CosmosDb/IGraphTraversalSource.Extensions.cs

```diff
@@ -56,27 +56,31 @@ public static class IGraphTraversalSourceExtensions
         }
 
         /// <summary>
-        /// Adds the "E()" step to the traversal.
+        /// Adds the "E().HasLabel(label_for_type_of_edge)" step to the traversal.
         /// </summary>
         /// <typeparam name="TEdge">The type of the edge.</typeparam>
         /// <param name="graphTraversalSource">The graph traversal source.</param>
         /// <param name="edgeIds">The edge id(s).</param>
         /// <returns>Returns the resulting traversal</returns>
         public static ISchemaBoundTraversal<Edge, TEdge> E<TEdge>(this IGraphTraversalSource graphTraversalSource, params object[] edgeIds)
         {
+            var label = LabelNameResolver.GetLabelName(typeof(TEdge));
+            var traversal = graphTraversalSource.E(edgeIds).HasLabel(label);
             return graphTraversalSource.E(edgeIds).AsSchemaBound<Edge, TEdge>();
```

Comment:
This line should be returning the `traversal` object now:
`return traversal.AsSchemaBound<Edge, TEdge>();`